plugins/search-backend-node/config.d.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  /** Configuration options for the search plugin */
+  search?: {
+    /**
+     * Search engine to use. Defaults to Lunr inmemory search engine
+     */
+    engine?: 'Lunr' | 'ElasticSearch';
```

Comment:
I would vote to do this only as code. Otherwise integrators can't add custom search engines since this type is not extendable. An alternative could be `config.has('search.elasticSearch')` if an integrator wants to e.g. use a different engine on localhost vs. in production.

Reply:
Yup, that would likely make sense to keep it more nimble :+1: